common/src/main/java/com/microsoft/identity/common/internal/cache/MsalOAuth2TokenCache.java

```diff
@@ -161,6 +162,182 @@ public void saveTokens(
         saveCredentials(accessTokenToSave, refreshTokenToSave, idTokenToSave);
     }
 
+    @Override
+    public Account getAccount(@NonNull final String environment,
+                              @NonNull final String clientId,
+                              @NonNull final String homeAccountId) {
+        final List<Account> allAccounts = getAccounts(environment, clientId);
+
+        // Return the sought Account matching the supplied homeAccountId
+        for (final Account account : allAccounts) {
+            if (homeAccountId.equals(account.getHomeAccountId())) {
+                return account;
+            }
+        }
+
+        return null;
+    }
+
+    @Override
+    public List<Account> getAccounts(@NonNull final String environment,
+                                     @NonNull final String clientId) {
+        final List<Account> accountsForThisApp = new ArrayList<>();
+
+        // Get all of the Accounts for this environment
+        final List<Account> accountsForEnvironment =
+                mAccountCredentialCache.getAccountsFilteredBy(
+                        null, // wildcard (*) homeAccountId
+                        environment,
+                        null // wildcard (*) realm
+                );
+
+        // Declare a List to hold the MicrosoftStsAccounts
+        final List<Account> microsoftStsAccounts = new ArrayList<>();
+
+        // Grab all of the Accounts supplied by either the v1 or v2 endpoint
+        for (final Account account : accountsForEnvironment) {
+            if (MicrosoftAccount.AUTHORITY_TYPE_V1_V2.equals(account.getAuthorityType())) {
+                microsoftStsAccounts.add(account);
+            }
+        }
+
+        // Grab the Credentials for this app...
+        final List<Credential> appCredentials =
```

Comment:
appCredentials is a potentially misleading name.... do you mean get all credentials currently in the cache?  If so... perhaps... allCredentials?

Reply:
Going to leave as-is; these credentials are loaded only for the current clientId (see param to `getCredentialsFilteredBy`)